operator/pkg/model/translation/envoy_listener_test.go

```diff
@@ -113,4 +126,18 @@ func TestNewSNIListener(t *testing.T) {
 		require.Len(t, listener.GetFilterChains(), 1)
 		require.Len(t, listener.GetFilterChains()[0].FilterChainMatch.ServerNames, 2)
 	})
+
+	t.Run("normal SNI listener with Proxy Protocol", func(t *testing.T) {
+		res, err := NewSNIListener("dummy-name", map[string][]string{"dummy-namespace/dummy-service:443": {"example.org", "example.com"}}, WithProxyProtocol())
+		require.Nil(t, err)
+
+		listener := &envoy_config_listener.Listener{}
+		err = proto.Unmarshal(res.Value, listener)
+		require.Nil(t, err)
+
+		require.Equal(t, "dummy-name", listener.Name)
+		require.Len(t, listener.GetListenerFilters(), 2)
```

Comment:
I know the test just above does this, but it would be good to check that the second filter is a Proxy Protocol filter here - that will protect us in case we add other filters in the future.